app/api/common.v2/database/MongoDataSource.ts

```diff
@@ -120,6 +189,44 @@ export abstract class MongoDataSource<CollectionSchema extends Document = any> {
     return this.collectionProxy;
   }
 
+  private async insertSyncLogs(mongoIds: ObjectId[]) {
+    if (mongoIds.length !== 0) {
+      await this.db.collection('updatelogs').insertMany(
+        mongoIds.map(insertedId => ({
+          timestamp: Date.now(),
+          namespace: this.collectionName,
+          mongoId: insertedId,
+          deleted: false,
+        })),
+        { session: this.getSession() }
+      );
+    }
+  }
+
+  private async upsertSyncLogs(
+    conditions: Filter<WithId<CollectionSchema>>[],
+    deleted: boolean = false
+  ) {
+    if (conditions.length === 0) {
+      return;
+    }
+
+    const modifiedDocuments = this.getCollection().find(
+      { $or: conditions },
+      { projection: { _id: 1 } }
+    );
+
+    await new MongoResultSet(modifiedDocuments, d => d).forEach(async ({ _id }) => {
+      await this.db.collection('updatelogs').updateMany(
```

Comment:
changed to a bulk write stream using the scoped collection without the updatelogs functionality, dont really love it since i had to change the type of the scope method to a generic and had to move the cache for the proxied collection away because the class now uses 2 different collections which leads me to think that maybe the best option is for the "updatelogs" collection to have its own datasource instance ?

Reply:
solved by changing proxy implementation to a collection wrapper